database/src/main/java/com/firebase/ui/database/OnChangedListener.java

```diff
@@ -0,0 +1,41 @@
+package com.firebase.ui.database;
+
+import com.google.firebase.database.ChildEventListener;
+import com.google.firebase.database.DataSnapshot;
+import com.google.firebase.database.DatabaseError;
+
+public interface OnChangedListener {
```

Comment:
I'm fine pulling this up. But wouldn't it be better as an interface in FirebaseArray (which I see moving towards becoming an abstract base-class or interface itself)? Having this inside FirebaseArray makes it more explicitly scoped as "these operations apply to this FirebaseArray (or subclass) object".